test/hack/resource/pkg/resourcetypes/instanceprofile.go

```diff
@@ -44,39 +44,40 @@ func (ip *InstanceProfile) Global() bool {
 }
 
 func (ip *InstanceProfile) GetExpired(ctx context.Context, expirationTime time.Time, excludedClusters []string) (names []string, err error) {
-	out, err := ip.iamClient.ListInstanceProfiles(ctx, &iam.ListInstanceProfilesInput{})
+	instanceProfiles, err := ip.getAllInstanceProfiles(ctx)
 	if err != nil {
 		return names, err
 	}
 
-	errs := make([]error, len(out.InstanceProfiles))
-	for i := range out.InstanceProfiles {
+	errs := make([]error, len(instanceProfiles))
+	for i := range instanceProfiles {
 		profiles, err := ip.iamClient.ListInstanceProfileTags(ctx, &iam.ListInstanceProfileTagsInput{
-			InstanceProfileName: out.InstanceProfiles[i].InstanceProfileName,
+			InstanceProfileName: instanceProfiles[i].InstanceProfileName,
 		})
 		if err != nil {
 			errs[i] = err
 			continue
 		}
 
-		clusterName, _ := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
+		clusterName, foundClusterName := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
```

Comment:
nit: I can't remember. Maybe not a huge deal, but I recalled that we said we were fine with not doing the "found" check since the result of `lo.FromPtr` was going to be an empty string anyways, which isn't valid for a cluster name or for a region name